### PR TITLE
feat: Add memory column

### LINKED ISSUE #204: Ability to see process memory footprint
## Feature Request
I would like to be able to see the memory footprint of running processes.

#### Use Case:
I want to use process-compose to assist local development of microservices. Sometimes I need several non-dockerized services running at the same time, and seeing memory footprint of individual processes would be a big help to me.

#### Proposed Change:
Add a memory column in the TUI

#### Who Benefits From The Change(s)?
People running long processes like APIs with a considerable memory footprint.

#### Alternative Approaches

### PR DESCRIPTION
**Pull request by themartdev**
I think psutil is a useful dependency to add, it opens up a lot of doors for cross-platform process instrumentation. We could potentially add think like CPU percentage and such. (although I'm not sure if it would make sense to squeeze CPU% in there)

Fixes #204

### CODE REVIEW

File: src/app/process.go [hunk 1]
```diff
@@ -471,6 +474,23 @@ func (p *Process) getStartTime() time.Time {
...
+		return 0
+	}
+	proc, err := process.NewProcess(int32(p.procState.Pid))
+	if err != nil {
+		log.Err(err).Msgf("Could not find process")
+		return -1
+	}
+	meminfo, err := proc.MemoryInfo()
+	if err != nil {
+		log.Err(err).Msgf("Error retrieving memory stats for process: %d", p.procState.Pid)
```
Comment (F1bonacc1) [inline src/app/process.go:488]: Can you please log the process name?
```golang
log.Err(err).
Str("process", p.getName()).
Int("pid", p.procState.Pid).
Msg("Error retrieving memory stats for process")
```

**Review by F1bonacc1 — COMMENTED** [review]
Looks great!
A few small comments.
Don't worry about the CI failing. `default.nix` will require a `vendorHash` update because of the new package.

**Review by F1bonacc1 — APPROVED** [review]
Great! Thanks!

**Comment by themartdev** [comment]
How do you fix the nix build? I'm not familiar with it :thinking:

**Comment by F1bonacc1** [comment]
> How do you fix the nix build? I'm not familiar with it 🤔

I will take care of that once I merge.

**Comment by F1bonacc1** [comment]
Squashed and Merged, thanks again!!! ❤️

**Pull request closed without merging**